### ISSUE TITLE
cannot build with react-native@0.73.0-rc.2

### ISSUE THREAD

**Issue by lafest** [microsoft/react-native-code-push#2601]
I cannot build my app with react-native@0.73.0-rc.2

because of https://github.com/facebook/react-native/pull/37256 this change, `import com.facebook.react.devsupport.DevInternalSettings;` makes error. so, I resolved this error with below patch.

```patch
	diff --git a/node_modules/react-native-code-push/android/app/src/main/java/com/microsoft/codepush/react/CodePush.java b/node_modules/react-native-code-push/android/app/src/main/java/com/microsoft/codepush/react/CodePush.java
index 923dd15..f5e4cc4 100644
--- a/node_modules/react-native-code-push/android/app/src/main/java/com/microsoft/codepush/react/CodePush.java
+++ b/node_modules/react-native-code-push/android/app/src/main/java/com/microsoft/codepush/react/CodePush.java
@@ -10,8 +10,8 @@ import com.facebook.react.ReactPackage;
 import com.facebook.react.bridge.JavaScriptModule;
 import com.facebook.react.bridge.NativeModule;
 import com.facebook.react.bridge.ReactApplicationContext;
-import com.facebook.react.devsupport.DevInternalSettings;
 import com.facebook.react.devsupport.interfaces.DevSupportManager;
+import com.facebook.react.modules.debug.interfaces.DeveloperSettings;
 import com.facebook.react.uimanager.ViewManager;

 import org.json.JSONException;
@@ -152,7 +152,7 @@ public class CodePush implements ReactPackage {
         if (instanceManager != null) {
             DevSupportManager devSupportManager = instanceManager.getDevSupportManager();
             if (devSupportManager != null) {
-                DevInternalSettings devInternalSettings = (DevInternalSettings)devSupportManager.getDevSettings();
+                DeveloperSettings devInternalSettings = devSupportManager.getDevSettings();
                 Method[] methods = devInternalSettings.getClass().getMethods();
                 for (Method m : methods) {
                     if (m.getName().equals("isReloadOnJSChangeEnabled")) {
```

But, I have no idea how to support 0.73 and also below versions. can anybody help this issue?

### Environment

* react-native-code-push version: 8.0.0
* react-native version: 0.73.0-rc.2
* iOS/Android/Windows version: Android 31
* Does this reproduce on a debug build or release build? debug
* Does this reproduce on a simulator, or only on a physical device? simulator

(The more info the faster we will be able to address it!)

**Comment by MikhailSuendukov** [comment]
Hello and thanks for contacting us, we are looking into this issue

**Comment by idan666** [comment]
Is there an update on the new React Native version support?
There are compatibility issues arising with Android 14, which require React Native to be upgraded to latest stable version.

**Comment by griarman** [comment]
any updates?

**Comment by RayKay91** [comment]
Also need a fix...

**Comment by Dallas62** [comment]
#2617

**Comment by DmitriyKirakosyan** [comment]
Fixed in version 8.2.0.
